nomad/deployment_endpoint.go

```diff
@@ -399,6 +399,20 @@ func (d *Deployment) List(args *structs.DeploymentListRequest, reply *structs.De
 		return structs.ErrPermissionDenied
 	}
 
+	if args.Filter != "" {
+		// Check for incompatible filtering.
+		// The namespace is set to default if empty before it reaches here, so
+		// we can't check for an empty string.
+		hasLegacyFilter := args.Prefix != "" ||
+			args.RequestNamespace() != structs.DefaultNamespace
```

Comment:
I think this check can also be done somewhere common to all endpoints that use `QueryOptions`, but I couldn't quite find where 🤔